src/main/java/com/aws/greengrass/ipc/SubscribeToCertificateUpdatesOperationHandler.java

```diff
@@ -0,0 +1,216 @@
+     * @param certificateManager certificate manager
+     * @param authorizationHandler authorization handler
+     */
+    public SubscribeToCertificateUpdatesOperationHandler(OperationContinuationHandlerContext context,
+                                                         CertificateManager certificateManager,
+                                                         AuthorizationHandler authorizationHandler) {
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.certificateManager = certificateManager;
+        this.authorizationHandler = authorizationHandler;
+
+    }
+
+    @Override
+    protected void onStreamClosed() {
+        certificateManager.unsubscribeFromServerCertificateUpdates(serverCertificateCallback);
+    }
+
+    /**
+     * Override to appropriately enforce stream responses are sent after the initial response.
+     * This only gets called if handleRequest returns normally and starts to send a response.
+     * <p>
+     * Here, we know that the initial SubscribeToCertificateUpdatesResponse response is sent. So, we send the
+     * first streaming response of CertificateUpdateEvent.
+     * </p>
+     */
+    @Override
+    public void afterHandleRequest() {
+        synchronized (firstStreamingEvent) {
+            if (subscriptionResponseSent.compareAndSet(false, true)) {
+                CertificateUpdateEvent event = firstStreamingEvent.get();
+                if (event != null) {
+                    sendStreamEvent(event);
+                }
+                firstStreamingEvent.set(null);
+            }
+        }
+    }
+
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    @Override
+    public SubscribeToCertificateUpdatesResponse handleRequest(
+            SubscribeToCertificateUpdatesRequest subscribeToCertificateUpdatesRequest) {
+        return translateExceptions(() -> {
+            try {
+                doAuthorizationForSubscribingToCertUpdates(serviceName);
+            } catch (AuthorizationException e) {
+                logger.atWarn().kv("error", e.getMessage()).kv(COMPONENT_NAME, serviceName).log(UNAUTHORIZED_ERROR);
+                throw new UnauthorizedError(e.getMessage());
+            }
+            try {
+                CertificateType certificateType = getCertificateTypeFromCertificateOptions(
+                        subscribeToCertificateUpdatesRequest.getCertificateOptions());
+                if (CertificateType.SERVER.equals(certificateType)) {
+                    final String csr = getCSRForServer();
+
+                    certificateManager.subscribeToServerCertificateUpdates(csr, serverCertificateCallback);
+                }
+            } catch (NoSuchAlgorithmException | IOException | OperatorCreationException
+                    | KeyStoreException | CsrProcessingException e) {
+                logger.atError().cause(e).log("Unable to create the certificate signing request");
```

Comment:
As per the message - Are all these exceptions coming from `getCSRForServer`? If that is the case, can you just scope down the try?